### PR TITLE
Add ability to pass hex color to the `borderColor` and `backgroundColor ` options

### LINKED ISSUE #33: Support more colors
Chalk now supports 16 million colors, we should support that in the `borderColor` and `backgroundColor` options by also accept a HEX color.

https://github.com/chalk/chalk#256-and-truecolor-color-support

### PR DESCRIPTION
**Pull request by ndelangen**
I want my borders pink, so I added that feature. fixes #33 

This also does upgrade `camelcase`, which removes node support <6
If you want that reverted, let me know.

I added docs & test & examples.

screen shot 2018-04-03 at 00 24 25

### CODE REVIEW

**Comment by ndelangen** [comment]
@sindresorhus Would love to have this released to be able to render pink borders everywhere.. Let me know if if you need help with maintenance on this package. ♥️

File: readme.md [hunk 1]
```diff
@@ -59,8 +59,9 @@ Type: `Object`
 
 ##### borderColor
 
-Type: `string`<br>
+Type: `string` or `function`<br>
 Values: `black` `red` `green` `yellow` `blue` `magenta` `cyan` `white` `gray`
+Value in the case of a function should be a `chalk.hex` or `chalk.rgb` or any other [chalk color models](https://github.com/chalk/chalk#256-and-truecolor-color-support)
```
Comment (sindresorhus) [inline readme.md]: I don't think we should expose everything like this. We can just allow a string with a color HEX.

**Comment by ndelangen** [comment]
I updated the PR @sindresorhus

**Comment by sindresorhus** [comment]
Looks good. Can you also update the docs and PR title?

Also fix the linter errors: https://travis-ci.org/sindresorhus/boxen/jobs/376248803

**Comment by ndelangen** [comment]
Strange I do not get that lint error locally.

Updated

**Comment by ndelangen** [comment]
So interestingly enough travis runs the script in a shell not supporting hex colours 😄

**Comment by ndelangen** [comment]
Looks like somewhere Node 4 support was dropped, possibly in `xo`?

```
> xo && nyc ava
/home/travis/build/sindresorhus/boxen/node_modules/xo/main.js:115
const {input, flags: opts} = cli;
      ^
SyntaxError: Unexpected token {
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/home/travis/build/sindresorhus/boxen/node_modules/xo/cli.js:15:2)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
```

**Comment by ndelangen** [comment]
@sindresorhus what do you think?

File: index.js [hunk 2]
```diff
@@ -60,7 +60,11 @@ const getBorderChars = borderStyle => {
 	return chars;
 };
 
-const getBackgroundColorName = x => camelCase('bg', x);
+// Color must either be a valid chalk color or a color hex ( #55aabb )
```
Comment (LitoMore) [inline index.js]: @ndelangen What about use [chalk-pipe](https://github.com/LitoMore/chalk-pipe) to solve colors?

We can just pass a string to get colors, and even more styles.

Comment (ndelangen) [reply index.js]: What other styles would make sense?

Comment (LitoMore) [reply index.js]: Combined style and CSS keyword styles:

```
#55aabb.bold
lightgrey
pink
```

> See doc here: https://github.com/LitoMore/chalk-pipe#valid-styles

Comment (sindresorhus) [reply index.js]: @LitoMore I don't want to overcomplicate it.

File: index.js [hunk 3]
```diff
@@ -60,7 +60,11 @@ const getBorderChars = borderStyle => {
...
+const isHex = color => color.match(/^#[0-f]{6}/i);
```
Comment (sindresorhus) [inline index.js]: Users will expect shorthands to work, so this needs to take into account `#fff` too.

Comment (ndelangen) [reply index.js]: sure

**Pull request merged**